### PR TITLE
Added Preload and Version Numbers

### PR DESCRIPTION
**Pull request by ayumihamsaki**
Not sure why the `Auth.htm` was missing the version numbers on the loaded assets (I added them into the code). I also `preloaded` all main assets.

This relates to the github issue: https://github.com/octobercms/october/issues/4271

### CODE REVIEW

**Comment by LukeTowers** [comment]
Could you add some performance numbers in here and explain how preload works / browser support for it?

**Comment by ayumihamsaki** [comment]
## Browser Support

- IE11 No
- Edge 17+
- Firefox 56+
- Chrome 50+
- Safari 11+
- Opera 37+
- Samsung Internet 5+

## What is Preload

Preload (spec) is a new web standard aimed at improving performance and providing more granular loading control to web developers. It gives developers the ability to define custom loading logic without suffering the performance penalty that script-based resource loaders incur.

## Performance (with and without Preload with a Waterfall Chart)

1_PSMeFcC3AXDUmdNf5l19Ug

### Without (Preload)

In the above image the assets are loaded when the Browser finds them.

### With (Preload)

In the above image the assets are loaded straight away by the Browser (reduced wait time)

## October CMS

With regards to October CMS the issue is that the `storm.min.js` file needs to be loaded before the rest of the assets and if it gets delayed then the browser can not load the rest of the file assets. When you add preload the assets are loaded before the rendering of the browser begins - hence the name **preload** (the assets are preloaded before the rendering begins).

---

Hope that explains and answers your question.

**Comment by LukeTowers** [comment]
I don't recognize `as="style"` as valid, what's the browser support for that? Also, does every single script need to be loaded in that way?

**Comment by ayumihamsaki** [comment]
> I don't recognize `as="style"` as valid, 

That is part of the Preload W3c spec see here for the spec details: https://www.w3.org/TR/preload/#as-attribute

It's a valid HTML attribute.

> what's the browser support for that?

That attribute will work fine in all browsers that support and don't support the `as` attribute. I know because I asked W3c that exact question in the past. So for example in IE11 that doesn't support that attribute, there won't be any errors and the browser just doesn't process the `link-rel preload` code.

> Also, does every single script need to be loaded in that way?

No, you should only preload the file assets that you are 100% sure they are going to be loaded when the user opens the web page.

For example, let's say I have a javascript that will run when the user scrolls down the web page. That shouldn't be preloaded and will show up in the console.log as a warning message saying:

`the following script was preloaded but not used within 1 second.`

Basically **I have preloaded only the assets that I know 100% will be loaded at the start of the browser render**.

There is a balance between preloading things and I have been using the preload spec for over 3 years and know which things need to be preloaded and which things will give a negative performance.

---

Happy to answer anymore questions on it. I'm almost an expert on that api, so can answer most questions on it.

**Comment by LukeTowers** [comment]
Thanks @ayumihamsaki.

I think I know why:
> Not sure why the Auth.htm was missing the version numbers on the loaded assets (I added them into the code)

If you have the version number in that then anyone would be able to visit the login page for an OctoberCMS website and determine the exact build number that was used. It used to be considered best practice to obscure the version / software you were using for security however that's increasingly pointless with automated scanning.
